core/src/org/javarosa/core/services/locale/LocalizationUtils.java

```diff
@@ -72,10 +72,12 @@ private static void parseAndAdd(OrderedHashtable locale, String line, int curlin
 
             //trim whitespace.
             line = line.trim();
+
+            int i = 0;
             
-            //clear comments
-            while(line.indexOf("#") != -1) {
-                line = line.substring(0, line.indexOf("#"));
+            //clear comments except if they have backslash before them (markdown '#'s)
+            while((i = line.indexOf("#")) != -1 && !(line.charAt(i-1) == '\\')) {
```

Comment:
if the "#" is at 0, this will crash.

We should add tests for parsing lines to the test suite for javarosa in general to catch everything